### PR TITLE
[MOD] Features requested for kick/ban.

### PR DESCRIPTION
**Pull request by Ianardo-DiCaprio**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Added features to kick and ban requested in #2649. This will now send an embed to the user when they are kicked/banned from a server detailing if it was a kick or ban, from what server, <s>the moderator that issued the kick/ban</s> and the reason for the kick/ban. Also blacked.

### CODE REVIEW

**Comment by yamikaitou** [comment]
Please allow this feature to be optional (default off)

File: redbot/cogs/mod/kickban.py [hunk 1]
```diff
@@ -179,6 +179,20 @@ class KickBanMixin(MixinMeta):
             return
         audit_reason = get_audit_reason(author, reason)
         try:
+            if reason == None:
```
Comment (NeuroAssassin) [inline redbot/cogs/mod/kickban.py:177]: Comparison with `None`s should be using `is`, not `==`.

File: redbot/cogs/mod/kickban.py [hunk 2]
```diff
@@ -179,6 +179,20 @@ class KickBanMixin(MixinMeta):
...
+                reason = "No reason was given."
+            else:
+                reason = reason
```
Comment (NeuroAssassin) [inline redbot/cogs/mod/kickban.py:180]: You don't need the `else` here for redefining the reason to itself.

File: redbot/cogs/mod/kickban.py [hunk 3]
```diff
@@ -179,6 +179,20 @@ class KickBanMixin(MixinMeta):
...
+            self == self.bot.get_cog("Mod")
```
Comment (NeuroAssassin) [inline redbot/cogs/mod/kickban.py:181]: Is this supposed to be a `=`?  And if so, `self` is already the Mod cog, so this wouldn't be needed.

File: redbot/cogs/mod/kickban.py [hunk 4]
```diff
@@ -218,7 +232,22 @@ class KickBanMixin(MixinMeta):
 
         If days is not a number, it's treated as the first word of the reason.
         Minimum 0 days, maximum 7. Defaults to 0."""
-
+        author = ctx.author
+        guild = ctx.guild
+        if reason == None:
```
Comment (NeuroAssassin) [inline redbot/cogs/mod/kickban.py:242]: Should use `is` instead of `==` when comparing with `None`.

File: redbot/cogs/mod/kickban.py [hunk 5]
```diff
@@ -218,7 +232,22 @@ class KickBanMixin(MixinMeta):
...
+            reason = "No reason was given."
+        else:
+            reason = reason
```
Comment (NeuroAssassin) [inline redbot/cogs/mod/kickban.py:245]: You don't need the `else` here for redefining `reason`.

File: redbot/cogs/mod/kickban.py [hunk 6]
```diff
@@ -218,7 +232,22 @@ class KickBanMixin(MixinMeta):
...
+        self == self.bot.get_cog("Mod")
```
Comment (NeuroAssassin) [inline redbot/cogs/mod/kickban.py:246]: Same thing as above, if its supposed to be a `=`, then it isn't needed as `self` is already the Mod cog.

File: redbot/cogs/mod/kickban.py [hunk 7]
```diff
@@ -179,6 +179,20 @@ class KickBanMixin(MixinMeta):
             return
         audit_reason = get_audit_reason(author, reason)
         try:
+            if reason == None:
+                reason = "No reason was given."
+            else:
+                reason = reason
+            self == self.bot.get_cog("Mod")
+            toggle = await self.settings.guild(guild).toggle_dm()
+            if toggle == True:
```
Comment (NeuroAssassin) [inline redbot/cogs/mod/kickban.py]: This can be changed to just `if toggle:`.

File: redbot/cogs/mod/kickban.py [hunk 8]
```diff
@@ -218,7 +232,22 @@ class KickBanMixin(MixinMeta):
...
-
+        author = ctx.author
+        guild = ctx.guild
+        if reason == None:
+            reason = "No reason was given."
+        else:
+            reason = reason
+        self == self.bot.get_cog("Mod")
+        toggle = await self.settings.guild(guild).toggle_dm()
+        if toggle == True:
```
Comment (NeuroAssassin) [inline redbot/cogs/mod/kickban.py]: Can be changed to just `if toggle:`.

**Review by NeuroAssassin — CHANGES_REQUESTED** [review]
I've made a list of things could use a few improvements/changes.  Here's the list:

File: redbot/cogs/mod/kickban.py [hunk 9]
```diff
@@ -218,7 +232,22 @@ class KickBanMixin(MixinMeta):
...
+            with contextlib.suppress(discord.HTTPException):
+                em = discord.Embed(
+                    title=_("**You have been banned from {guild}.**").format(guild=guild)
+                )
+                em.add_field(name=_("**Moderator**"), value=author.name, inline=False)
```
Comment (mikeshardmind) [inline redbot/cogs/mod/kickban.py]: Showing the reason here is fine. Showing the moderator is not unless there is another setting for this.

File: redbot/cogs/mod/settings.py [hunk 10]
```diff
@@ -199,3 +203,20 @@ class ModSettings(MixinMeta):
                     command=f"{ctx.prefix}unban"
                 )
             )
+
+    @modset.command()
+    @commands.guild_only()
+    async def toggledm(self, ctx: commands.Context):
```
Comment (mikeshardmind) [inline redbot/cogs/mod/settings.py]: For consistency with other core commands with configuration, this should accept a bool for setting it directly. If you'd like it to act as a toggle when that bool is not provided, that is fine.

**Review by mikeshardmind — CHANGES_REQUESTED** [review]
In addition to some feedback another contributor pointed out, there are a few issues of design here.

File: redbot/cogs/mod/kickban.py (hunk 9 above)
Comment (Ianardo-DiCaprio) [reply redbot/cogs/mod/kickban.py]: Would you prefer no moderator or an extra setting for it?

**Pull request closed without merging**